### ISSUE TITLE
Empty selections cannot be composed with "and" in a transform filter

### ISSUE THREAD

**Issue by victornoel** [vega/vega-lite#5553]
Here is an [example](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAkDODGALApgWwIaQFxUQFzwAdYsB6UgN2QHN0A6agSz0QFcAjOxge1IRQyUa6ALQAbZskoBmOgBY6ABjoArWNwB2kADTgoAE3R5MOUBAiRWAJzHZcBYmSG0GzNpx7PRh47GR5YUh90Unh0K1hVdS09AF9dC2QNeG59Rg1qOzNzSBSxbissvXMoADNGZDF9O0gAeStGJi0Ekqg8AE9CZBqNblR09FtisHjhyDxubjE8RkI7AG1hiGzWsoqqmoA5dFRultXxzu6cSF7+jUHIJZH98xXWyHLK6pP6xvSda4sOrp6+gaGq1Gq3uJUe6xeUAAEgU-IRuAB3ZCFW5gn7HKAAR1Y6A0M2MMyoVyBqLAoJyTw2JwAsowxMhYAB9LpWRkAcUG+WaXzaRxq2Nx+KMjCJ11iwwAurdIAAPIoHSmQyAwiLIeFIlHXQ6-E4CvHMYWikrAnLteUPRU1Wn0pks9mczSfA7o-k4-UEkXdYbiiAmyAYKwAa3N3z5J3gjCs8HpxJuekgfnp8BmjtMY0TyGTBUUAEYcyGcux0pDyRZ0oRWHgahnk07VlBuIQUxpYAseRANKwxGJSWCAKoAZQAgnX61AAKLWRt7dtQABS6EIuNjqwlYtJ4OerZwizHdQaTRX5ilWpdJ1g6WoMe90prEysuYATAWLEWNCWeZBy5Xq5VM1XewbJseBbNsxw7Lse1nAAOQCIDkOCwAANkQ6REIAVh5NcSS1RVtzAXd60gABhdoJHfZFW2uE9nTDKALwya9jTiFpxisXFYFKApUDAu5cLpPBkRfHI7xA4SwVxSFCPA+i-yzB881HcCEzk+8nyPVceR9VptN9PQ11iIA)

When you select:
- null (aka empty selection if I understood well), the whole dataset is kept
- a value for only one of the select input, the chart gets blank
- a value for both inputs, the filter is applied

I would expect that selecting a value for only one of the two inputs would filter according to this input.

Note that the same problem happens:
- without using the null value but only at first (because before the user touch the input, the selection is considered empty in the same way)
- when using multiple fields and bindings (select) in the same selection, but in that case, instead of getting a blank chart, we get all points.

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v3.4.0.json",
  "data": {
    "url": "https://vega.github.io/vega-datasets/data/cars.json"
  },
  "encoding": {
    "color": {
      "field": "Origin",
      "type": "nominal"
    },
    "tooltip": [
      {
        "field": "Name",
        "type": "nominal"
      },
      {
        "field": "Origin",
        "type": "nominal"
      },
      {
        "field": "Horsepower",
        "type": "quantitative"
      },
      {
        "field": "Miles_per_Gallon",
        "type": "quantitative"
      }
    ],
    "x": {
      "field": "Horsepower",
      "type": "quantitative"
    },
    "y": {
      "field": "Miles_per_Gallon",
      "type": "quantitative"
    }
  },
  "mark": {
    "type": "circle"
  },
  "selection": {
    "selector011": {
      "bind": {
        "input": "select",
        "options": [
          null,
          "USA",
          "Europe",
          "Japan"
        ]
      },
      "fields": [
        "Origin"
      ],
      "type": "single"
    },
    "selector012": {
      "bind": {
        "input": "select",
        "options": [
          null,
          8,
          4,
          6,
          3,
          5
        ]
      },
      "fields": [
        "Cylinders"
      ],
      "type": "single"
    }
  },
  "transform": [
    {
      "filter": {
        "selection": {
          "and": [
            "selector011",
            "selector012"
          ]
        }
      }
    }
  ]
}
```

**Comment by petermattia** [comment]
FYI - I have a toy example of this bug in [this SO question](https://stackoverflow.com/questions/58919888/multiple-selections-in-altair/58920296?noredirect=1#comment104106712_58920296)

**Comment by jeremysalwen** [comment]
I am seeing this bug as well.  Is there any workaround?

**Comment by drgould** [comment]
Moving the "and" outside of the selection is a workaround. Like so:
```json
{
  "filter": {
    "and": [
      {"selection": "selection_1"},
      {"selection": "selection_2"}
    ]
  }
}
```

**Comment by arvind** [comment]
Thank you @victornoel for filing the issue, and @drgould for the workaround. This issue will be fixed with #7155, which standardizes the workaround by dropping support for logical compositions _within_ a single `"selection"` property.
